code/include/playfab_c/PlayFabClientDataModels_c.h

```diff
@@ -0,0 +1,515 @@
+    GF,
+    PF,
+    TF,
+    GA,
+    GM,
+    GE,
+    DE,
+    GH,
+    GI,
+    GR,
+    GL,
+    GD,
+    GP,
+    GU,
+    GT,
+    GG,
+    GN,
+    GW,
+    GY,
+    HT,
+    HM,
+    VA,
+    HN,
+    HK,
+    HU,
+    IS,
+#pragma push_macro("IN")
+#undef IN
+    IN,
+#pragma pop_macro("IN")
+    ID,
+    IR,
+    IQ,
+    IE,
+    IM,
+    IL,
+    IT,
+    JM,
+    JP,
+    JE,
+    JO,
+    KZ,
+    KE,
+    KI,
+    KP,
+    KR,
+    KW,
+    KG,
+    LA,
+    LV,
+    LB,
+    LS,
+    LR,
+    LY,
+    LI,
+    LT,
+    LU,
+    MO,
+    MK,
+    MG,
+    MW,
+    MY,
+    MV,
+    ML,
+    MT,
+    MH,
+    MQ,
+    MR,
+    MU,
+    YT,
+    MX,
+    FM,
+    MD,
+    MC,
+    MN,
+    ME,
+    MS,
+    MA,
+    MZ,
+    MM,
+    NA,
+    NR,
+    NP,
+    NL,
+    NC,
+    NZ,
+    NI,
+    NE,
+    NG,
+    NU,
+    NF,
+    MP,
+    NO,
+    OM,
+    PK,
+    PW,
+    PS,
+    PA,
+    PG,
+    PY,
+    PE,
+    PH,
+    PN,
+    PL,
+    PT,
+    PR,
+    QA,
+    RE,
+    RO,
+    RU,
+    RW,
+    BL,
+    SH,
+    KN,
+    LC,
+    MF,
+    PM,
+    VC,
+    WS,
+    SM,
+    ST,
+    SA,
+    SN,
+    RS,
+    SC,
+    SL,
+    SG,
+    SX,
+    SK,
+    SI,
+    SB,
+    SO,
+    ZA,
+    GS,
+    SS,
+    ES,
+    LK,
+    SD,
+    SR,
+    SJ,
+    SZ,
+    SE,
+    CH,
+    SY,
+    TW,
+    TJ,
+    TZ,
+    TH,
+    TL,
+    TG,
+    TK,
+    TO,
+    TT,
+    TN,
+    TR,
+    TM,
+    TC,
+    TV,
+    UG,
+    UA,
+    AE,
+    GB,
+    US,
+    UM,
+    UY,
+    UZ,
+    VU,
+    VE,
+    VN,
+    VG,
+    VI,
+    WF,
+    EH,
+    YE,
+    ZM,
+    ZW
+};
+
+enum class PlayFabSubscriptionProviderStatus : uint32_t
+{
+    NoError = 0,
+    Cancelled = 1,
+    UnknownError = 2,
+    BillingError = 3,
+    ProductUnavailable = 4,
+    CustomerDidNotAcceptPriceChange = 5,
+    FreeTrial = 6,
+    PaymentPending = 7
+};
+
+enum class PlayFabPushNotificationPlatform : uint32_t
+{
+    ApplePushNotificationService = 0,
+    GoogleCloudMessaging = 1
+};
+
+struct PlayFabPlayerProfileViewConstraints
+{
+    bool showAvatarUrl;
+    bool showBannedUntil;
+    bool showCampaignAttributions;
+    bool showContactEmailAddresses;
+    bool showCreated;
+    bool showDisplayName;
+    bool showExperimentVariants;
+    bool showLastLogin;
+    bool showLinkedAccounts;
+    bool showLocations;
+    bool showMemberships;
+    bool showOrigination;
+    bool showPushNotificationRegistrations;
+    bool showStatistics;
+    bool showTags;
+    bool showTotalValueToDateInUsd;
+    bool showValuesToDate;
+};
+
+struct PlayFabGetPlayerCombinedInfoRequestParams
+{
+    bool getCharacterInventories;
+    bool getCharacterList;
+    bool getPlayerProfile;
+    bool getPlayerStatistics;
+    bool getTitleData;
+    bool getUserAccountInfo;
+    bool getUserData;
+    bool getUserInventory;
+    bool getUserReadOnlyData;
+    bool getUserVirtualCurrency;
+    const char** playerStatisticNames;
+    size_t playerStatisticNamesCount;

```

Comment:
should this be `char const * const *` (alternatively `const PCSTR*`)?

that way someone who receives access to this struct can't overwrite the pointer value being pointed to (not I'm not refererring to the char data underneath that pointer)

it's really hard to talk about double pointers with precise language so here's a simplified example which demonstrates what I'm talking about

i.e.
```
#include <cstdio>

struct TestStruct
{
    const char** strptr;
};

void func(
    const TestStruct& ts // should only be able to read values in this struct
    )
{
    *(ts.strptr) = nullptr; // uhoh I can write data where I shouldn't be able to.
}

void main()
{
    const char* str = "test";
    const TestStruct ts{ &str };
    func(ts);
    printf("str = 0x%p\n", str); // prints "str = 0x0000000" (null). func was able to modify data on the test struct
}
```

this might also apply to other double pointers